src/mem/allocconfig.h

```diff
@@ -4,12 +4,6 @@
 
 namespace snmalloc
 {
-// The CHECK_CLIENT macro is used to turn on minimal checking of the client
-// calling the API correctly.
-#if !defined(NDEBUG) && !defined(CHECK_CLIENT)
-#  define CHECK_CLIENT
-#endif
-
```

Comment:
This hunk seems either unrelated or mistaken, I amn't sure which.  Do you mean for `CHECK_CLIENT` to be completely determined by `cmake` now?

Reply:
It should probably be another PR.  I really want to make 

*   Secure for users of snmalloc
*   Debug for developers for snmalloc

These are different and switching on checking in Debug using encrypted pointers is not the easiest to debug issues.  We build all the tests with the checked flag on.  But everything checked is checking is a user error, not a snmalloc error.